src/draw/lv_image_decoder.c

```diff
@@ -78,100 +86,52 @@ void _lv_image_decoder_deinit(void)
 
 lv_result_t lv_image_decoder_get_info(const void * src, lv_image_header_t * header)
 {
-    lv_memzero(header, sizeof(lv_image_header_t));
-
-    if(src == NULL) return LV_RESULT_INVALID;
-
-    lv_image_src_t src_type = lv_image_src_get_type(src);
-    if(src_type == LV_IMAGE_SRC_VARIABLE) {
-        const lv_image_dsc_t * img_dsc = src;
-        if(img_dsc->data == NULL) return LV_RESULT_INVALID;
-    }
+    lv_image_decoder_t * decoder = image_decoder_get_info(src, header);
+    if(decoder == NULL) return LV_RESULT_INVALID;
 
-    lv_result_t res = LV_RESULT_INVALID;
-    lv_image_decoder_t * decoder;
-    _LV_LL_READ(img_decoder_ll_p, decoder) {
-        if(decoder->info_cb) {
-            res = decoder->info_cb(decoder, src, header);
-            if(res == LV_RESULT_OK) {
-                if(header->stride == 0) header->stride = img_width_to_stride(header);
-                break;
-            }
-        }
-    }
-
-    return res;
+    return LV_RESULT_OK;
 }
 
 lv_result_t lv_image_decoder_open(lv_image_decoder_dsc_t * dsc, const void * src, const lv_image_decoder_args_t * args)
 {
     lv_memzero(dsc, sizeof(lv_image_decoder_dsc_t));
 
     if(src == NULL) return LV_RESULT_INVALID;
-    lv_image_src_t src_type = lv_image_src_get_type(src);
-    if(src_type == LV_IMAGE_SRC_VARIABLE) {
-        const lv_image_dsc_t * img_dsc = src;
-        if(img_dsc->data == NULL) return LV_RESULT_INVALID;
-    }
+    dsc->src_type = lv_image_src_get_type(src);
 
-    dsc->src_type = src_type;
-
-    if(dsc->src_type == LV_IMAGE_SRC_FILE) {
-        dsc->src = lv_strdup(src);
-    }
-    else {
-        dsc->src = src;
-    }
+#if LV_CACHE_DEF_SIZE > 0
+    dsc->cache = img_cache_p;
+    /*
+     * Check the cache first
+     * If the image is found in the cache, just return it.*/
+    if(try_cache(dsc) == LV_RESULT_OK) return LV_RESULT_OK;
+#endif
 
-    lv_result_t res = LV_RESULT_INVALID;
+    /*Find the decoder that can open the image source, and get the header info in the same time.*/
+    dsc->decoder = image_decoder_get_info(src, &dsc->header);
+    if(dsc->decoder == NULL) return LV_RESULT_INVALID;
 
-    lv_image_decoder_t * decoder;
+    /*Duplicate the source if it's a file*/
+    if(dsc->src_type == LV_IMAGE_SRC_FILE) dsc->src = lv_strdup(src);
+    else dsc->src = src;
 
-    static const lv_image_decoder_args_t def_args = {
+    /*Make a copy of args*/
+    dsc->args = args ? *args : (lv_image_decoder_args_t) {
```

Comment:
```c
    static const lv_image_decoder_args_t def_args = {
        .stride_align = LV_DRAW_BUF_STRIDE_ALIGN != 1,
        .premultiply = false,
        .no_cache = false,
        .use_indexed = false,
    };

    /*Make a copy of args */
    dsc->args = args ? *args : def_args;
```

See 
https://github.com/lvgl/lvgl/pull/4907
https://stackoverflow.com/questions/33270731/error-c4576-in-vs2015-enterprise/33271165#33271165

Reply:
I think it should work if the CI passes. Let's update it if it causes issues somewhere.